travis/setup_dependencies_common.sh

```diff
@@ -37,14 +37,51 @@ function retry_on_known_error() {
         echo "ERROR: Function retry_on_known_error() called without arguments." 1>&2
         return 1
     fi
+    _tmp_output_file="tmp.txt"
     _n_retries=0
     _exitval=0
     _retry=true
     while $_retry; do
         _retry=false
-        # Execute the wrapped command and get its unified output:
-        _output=$($@ 2>&1)
-        _exitval="$?"
+        # Execute the wrapped command and get its unified output.
+        # This command needs to run in the current shell/environment in case
+        # it sets environment variables (like 'conda install' does)
+        #
+        # Wow, bash is fussy about order. Redirection to a file needs to
+        # happen *before* the redirection of stderr to stdout
+        $@ 2>&1 | tee $_tmp_output_file
+        _exitval="${PIPESTATUS[0]}"
```

Comment:
@djhoese -- you should also change `$?` to this, otherwise `tee` is the last process and always succeeds (or so I'm told by stack overflow 🤣 )

Reply:
Ah neat. I will probably rebase any of my changes on your branch to double check that my project can pass with what's been done here so far.